docs/localdev.md

```diff
@@ -86,3 +86,33 @@ We will address two modes of working with the ExternalDNS operator
 ### Working with external Kuberenets/OpenShift clusters
 
 If you have any other cluster you'd like to test the operator on you just need to set KUBECONFIG and follow the instructions layed out in both sections indicated above
+
+## Kind (Kubernetes in Docker)
+This section walks you through the deployment of the external-dns-operator in a Kind environment.
+### Prerequisites
+- [Go](https://golang.org/doc/install#install) and [Docker](https://docs.docker.com/engine/install/) are installed.
+- [Kind](https://kind.sigs.k8s.io/docs/user/quick-start/#installation) is installed and a local Kubernetes [cluster](https://kind.sigs.k8s.io/docs/user/quick-start/#creating-a-cluster) is created.
+- [kubectl](https://kubernetes.io/docs/tasks/tools/#kubectl) is installed to access the above created cluster.
+- Any registry that is accessible from your cluster to store and distribute the docker images. [quay.io](https://quay.io/) is shown as an example here.
+### Build and Push the docker image
+1. Clone the [external-dns-operator](https://github.com/openshift/external-dns-operator) repo to your local machine's `$GOPATH/src`.
+2. Change directory `cd $GOPATH/src/external-dns-operator`
+3. Perform the `docker login` and provide the user credentials to access the registry incase of using the quay.io
+4. Build and push the external-dns-operator docker image by providing the relevant registry name and tag.
+   ```bash
+   # Docker image name can be provided using the $IMG variable,
+   $ export IMG=quay.io/<username>/external-dns-operator:latest
+   # BUILD the docker image
+   $ make image-build
+   # Push the built image to the registry
+   $ make image-push
+   ```
+### Deploy the external-dns-operator
```

Comment:
Should we give an example of how to use kubectl config, or set the KUBECONFIG envar. I have had the situation where KUBECONFIG was set to another cluster - I started Kind  but the `make deploy`  was looking for the version set by KUBECONFIG